### ISSUE TITLE
Update the standard library search paths

### ISSUE THREAD

**Issue by SuperFola** [ArkScript-lang/Ark#310]
On Linux, we should search for the standard library under `/usr/lib` and `/usr/local/lib`, otherwise use the `ARKSCRIPT_PATH` environment variable.

On Windows, we should search under `C:/Program Files/Ark`, otherwise use the environment variable.

**Details**

The environment variable `ARKSCRIPT_PATH` should point to a folder containing the standard library:

```
ARKSCRIPT_PATH/
|__ std/
    |__ String.ark
    |__ List.ark
    |__ ...
|__ modules/
|__ ...
```

----

Checklist
- [ ] update the website documentation accordingly
- [ ] update the readme if needed
- [ ] update the doxydoc if needed
- [ ] should work without the environment variable `ARKSCRIPT_PATH`
- [ ] should work with the environment variable

**Comment by madstk1** [comment]
Would it not make more sense to act on the LD_LIBRARY_PATH variable in Linux instead?

**Comment by SuperFola** [comment]
Wouldn't this be a bit counter intuitive since `LD_LIBRARY_PATH` is for dynamically loaded libraries (such as .so)? I mean, this is an interesting idea, though this might create conflicts when introducing the virtual environments (#256).

**Comment by madstk1** [comment]
In that case, would it make more sense to make ARKSCRIPT_PATH a delimitered list of search paths, like PATH or LD_LIBRARY_PATH? I don't know much about virtual environments, so this might not work. It would at least add some extensibility, for future.

**Comment by SuperFola** [comment]
I understand the idea. The point of ARKSCRIPT_PATH is to be able to work with and without a venv. The problem I see with going full LD_LIBRARY_PATH is that it won't work on non-Unix systems, such as Windows. However, I like the idea of providing a list of search paths, we'll have to see how that can be integrated with the CMake install rules (or we could just define ARKSCRIPT_PATH as a list like `/usr/lib/arkscript;/usr/local/lib/arkscript;/home/myuser/myarkscript/blabla` and this would work for global installations and local ones).

----

Basically a venv is just
- save the PATH, ARKSCRIPT_PATH
- update the PATH (to get to the venv arkscript binary)
- update the ARKSCRIPT_PATH (to get the venv arkscript std library)

**Comment by SuperFola** [comment]
This most likely would have to be defined at runtime through an environment variable (eg ask the user to create it or have a script doing the work when installing).

Feel free to hop to the discord if you need any help with the project structure!
